server/resource_server/service/service.go

```diff
@@ -172,7 +179,54 @@ func (s *service) ProfileClient(userName string) (models.ClientResponseOutput, e
 }
 
 func (s *service) VerifyEmail(userID uint) error {
-	return s.repository.VerifyEmail(userID)
+	user, e := s.repository.FindUser(userID)
+	if e != nil {
+		return e
+	}
+
+	verificationCode := s.emailVerifier.GenerateVerificationCode(&user)
+
+	e = s.emailVerifier.SendVerificationEmail(&user, verificationCode)
+	if e != nil {
+		return e
+	}
+
+	e = s.repository.SaveEmailVerificationData(
+		models.EmailVerificationData{
+			UserId:           user.ID,
+			VerificationCode: verificationCode,
+			ExpiresAt:        time.Now().Add(constants.VerificationCodeValidityDuration),
+		},
+	)
+
+	return e
+}
+
+func (s *service) VerifyCode(userId uint, code string) error {
+	verificationData, e := s.repository.GetEmailVerificationData(userId)
+	if e != nil {
+		return e
+	}
+
+	e = s.emailVerifier.VerifyCode(&verificationData, code)
+	if e != nil {
+		return e
+	}
+
+	// generate zk-proof of the verification procedure
+	proof := "mock_proof"
+
+	e = s.repository.SaveProofOfEmailVerification(userId, code, proof)
+	if e != nil {
+		return e
+	}
+
+	e = s.repository.DeleteEmailVerificationData(userId)
```

Comment:
why we delete the data?

Reply:
We can delete this data because it's no longer used after the code is successfully verified.